src/test/transaction_tests.cpp

```diff
@@ -149,17 +154,20 @@ unsigned int FillFlags(unsigned int flags)
 
     // WITNESS implies P2SH (and transitively CLEANSTACK implies P2SH)
     if (flags & SCRIPT_VERIFY_WITNESS) flags |= SCRIPT_VERIFY_P2SH;
+    Assert(IsValidFlagCombination(flags));
     return flags;
 }
 
-// Return valid flags that are all except one flag for each flag
-std::vector<unsigned int> ExcludeIndividualFlags(unsigned int flags)
+// Exclude each possible script verify flag from flags. Returns a list of these flag combinations
```

Comment:
This returns a set, not a list.

You could copy the set into a vector to avoid changing the return type if you really felt like it.

```diff
diff --git a/src/test/transaction_tests.cpp b/src/test/transaction_tests.cpp
index 3d6ad20b85..235862ec7a 100644
--- a/src/test/transaction_tests.cpp
+++ b/src/test/transaction_tests.cpp
@@ -161,7 +161,7 @@ unsigned int FillFlags(unsigned int flags)
 // Exclude each possible script verify flag from flags. Returns a list of these flag combinations
 // that are valid and without duplicates. For example: if flags=1111 and there are 4 possible flags,
 // this should return the vector [0111, 1011, 1101, 1110].
-std::set<unsigned int> ExcludeIndividualFlags(unsigned int flags)
+std::vector<unsigned int> ExcludeIndividualFlags(unsigned int flags)
 {
     std::set<unsigned int> flags_combos;
     for (unsigned int i = 0; i < mapFlagNames.size(); ++i) {
@@ -170,7 +170,7 @@ std::set<unsigned int> ExcludeIndividualFlags(unsigned int flags)
             flags_combos.insert(flags_excluding_i);
         }
     }
-    return flags_combos;
+    return {flags_combos.begin(), flags_combos.end()};
 }
 
 BOOST_FIXTURE_TEST_SUITE(transaction_tests, BasicTestingSetup)
```

Reply:
True, although hopefully it's obvious - will update if I need to retouch. Thanks!

Reply:
Fixed